core/txpool/legacypool/cache_for_miner.go

```diff
@@ -0,0 +1,136 @@
+package legacypool
+
+import (
+	"math/big"
+	"sort"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/txpool"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+var (
+	pendingCacheGauge = metrics.NewRegisteredGauge("txpool/legacypool/pending/cache", nil)
+	localCacheGauge   = metrics.NewRegisteredGauge("txpool/legacypool/local/cache", nil)
+)
+
+// copy of pending transactions
+type cacheForMiner struct {
+	txLock   sync.Mutex
+	pending  map[common.Address]map[*types.Transaction]struct{}
+	locals   map[common.Address]bool
+	addrLock sync.Mutex
+}
+
+func newCacheForMiner() *cacheForMiner {
+	return &cacheForMiner{
+		pending: make(map[common.Address]map[*types.Transaction]struct{}),
+		locals:  make(map[common.Address]bool),
+	}
+}
+
+func (pc *cacheForMiner) add(txs types.Transactions, signer types.Signer) {
```

Comment:
Will pass by pointer(`txs *types.Transactions`) be more efficient?

Reply:
The `types.Transactions` is defined as ```type Transactions []*Transaction ```, so it doesn't take too much memory and should be ok to use `types.Transactions` directly.